src/flags/apiVersion.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Flags, CliUx } from '@oclif/core';
+import { Messages, sfdc } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/sf-plugins-core', 'messages');
+
+// versions below this are retired
+export const minValidApiVersion = 21;
+// this and all un-retired versions below it are deprecated
+export const maxDeprecated = 30;
+export const maxDeprecatedUrl = 'https://help.salesforce.com/s/articleView?id=000354473&type=1;';
+
+/**
+ * apiVersion for a salesforce org's rest api.
+ * Will validate format and that the api version is still supported.
+ * Will default to the version specified in Config, if it exists (and will provide an override warning)
+ *
+ * CAVEAT: unlike the apiversion flag on sfdxCommand, this does not set the version on the org/connection
+ * We leave this up to the plugins to implement
+ */
+export const apiVersionFlag = Flags.build<string>({
```

Comment:
this should be `orgApiVersionFlag` (or something like that) since we'll inevitably have to support an `--api-version` flag for something that's not the salesforce core api